src/H5Oefl.c

```diff
@@ -111,64 +103,69 @@ H5O__efl_decode(H5F_t *f, H5O_t H5_ATTR_UNUSED *open_oh, unsigned H5_ATTR_UNUSED
 
     /* Number of slots */
     UINT16DECODE(p, mesg->nalloc);
-    HDassert(mesg->nalloc > 0);
+    if (mesg->nalloc <= 0)
+        HGOTO_ERROR(H5E_OHDR, H5E_CANTLOAD, NULL, "bad number of allocated slots when parsing efl msg")
```

Comment:
Parsing a malformed file should never result in an assert() call.

Reply:
Agree

However, it's an unsigned value, so can't be <0

Reply:
I could change that to be != 0, but then that would be broken if anyone changes the type to be signed so they can have a "bad" value.